.github/workflows/coverity.yml

```diff
@@ -0,0 +1,110 @@
+# *******************************************************************************
+# Copyright (C) 2025 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+#
+# SPDX-License-Identifier: Apache-2.0
+# *******************************************************************************
+
+name: Coverity Scan
+
+on:
+  # Only run on push to main branch
+  push:
+    branches: [develop]

```

Comment:
We can keep `workflow_dispatch:` for easier debugging, but I don't have strong opinions on this.